### PR TITLE
feat(eds): add parser and queries for EDS files

### PR DESCRIPTION
**Pull request by uyha**

### CODE REVIEW

File: queries/eds/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,63 @@
+["="] @punctuation.delimeter
+
+(
+ (section_name) @variable.builtin
+ (#any-of? @variable.builtin "[FileInfo]" "[DeviceInfo]" "[DummyUsage]" "[MandatoryObjects]" "[OptionalObjects]")
+)
```
Comment (amaanq) [inline queries/eds/highlights.scm:6]: I can be wrong and maybe it's intended to have section_name include the brackets, but extracting the brackets to the outer node which is then followed by the section_name node of just text could be nicer (not an issue here)

File: queries/eds/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,63 @@
+["="] @punctuation.delimeter
```
Comment (amaanq) [inline queries/eds/highlights.scm:1]:
```suggestion
"=" @punctuation.delimeter
```

File: queries/eds/folds.scm [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[
+ (section)
+] @fold
```
Comment (amaanq) [inline queries/eds/folds.scm:3]:
```suggestion
(section) @fold
```

File: queries/eds/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+(
+ (section_name) @variable.builtin
+ (#any-of? @variable.builtin "[FileInfo]" "[DeviceInfo]" "[DummyUsage]" "[MandatoryObjects]" "[OptionalObjects]")
+)
+
+(
+ (section_name) @variable.builtin
+ (#lua-match? @variable.builtin "^%[1")
+)
```
Comment (amaanq) [inline queries/eds/highlights.scm:11]: Can we follow the style seen in pretty much every query? No dangling paren mainly..
```suggestion
((section_name) @variable.builtin
  (#lua-match? @variable.builtin "^%[1"))
```

File: queries/eds/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+(
+ (section
+   (section_name) @_name
+   (#lua-match? @_name "Comments")
+   ) @comment
+ )
```
Comment (amaanq) [inline queries/eds/highlights.scm:18]: This marks the whole section itself as a comment, not the name/body (or other nodes that are children), just want to double check that that's intended, and also the extra parenthesis pair isn't needed here and in a bunch of places below, would be nice to tidy that up

```suggestion
(section
   (section_name) @_name
   (#lua-match? @_name "Comments")) @comment
```

**Review by amaanq — CHANGES_REQUESTED** [review]

File: queries/eds/highlights.scm (hunk 5 above)
Comment (uyha) [reply queries/eds/highlights.scm:18]: yes, this is intentional that the whole section is a comment block, not just the section name.

**Review by amaanq — APPROVED** [review]

**Pull request merged**